New module: Add notification module for Office 365 Connector Cards (notification/office_365_connector_card)

marc-sensenich opened the pull request:
##### SUMMARY
Using the Incoming Webhook connector, create Office 365 Connector Cards

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
Office 365 Connector Cards

##### ANSIBLE VERSION
```
ansible 2.2.2.0
  config file =
  configured module search path = Default w/o overrides
```

##### ADDITIONAL INFORMATION
Similar to PR #22215, but covers all functionality as referenced by the [API](https://dev.outlook.com/Connectors/reference)

gundalow commented on lib/ansible/modules/notification/office_365_connector_card.py:
```diff
@@ -0,0 +1,225 @@
+module: office_365_connector_card
+short_descriptions: Use webhooks to create Connector Card messages within an Office 365 group
+description:
+  - The C(office_365_connector_card) module creates Connector Card messages through Office 365 Connectors U(https://dev.outlook.com/Connectors)
```
No need to repeat the module name just say 
`Creates Connector Cards... `

gundalow commented on lib/ansible/modules/notification/office_365_connector_card.py:
```diff
@@ -0,0 +1,225 @@
+options:
+  webhook:
+    description:
+      - The webhook URL is given to you when you create a new Connector
```
Missing full stop.

gundalow reviewed (changes requested):
Thanks for the new module
1) There are a few pep8 warning to clear up. 
2) Is this module idempotent, I don't see check mode defined. 
3) Can you remove cards? If so adding state=absent, present may be useful.

marc-sensenich commented:
Thanks for reviewing this.

1. Resolved in latest update
2. The actions wouldn't be idempotent, so check mode would not be present.
3. Currently the webhook does not allow delete, therefore it is create only

gundalow commented:
Could you please add a `note:` to say that this module isn't idempotent.

I assume it's safe to run the same playbook twice, or will you get two Connector Cards created?

marc-sensenich commented:
I added a note to say that it is not idempotent and will cause 2 cards to be created. I added the check mode in similar to how other notification modules have handled it as well.

gundalow commented:
Please work with the creator of https://github.com/ansible/ansible/pull/22215 to decide which module should go forward. You can both me authors & maintainers of the unified module.

marc-sensenich commented:
Any thoughts on moving this forward. I see there hasn't been a response from the owner of #22215

marc-sensenich commented:
Closed this one, will reopen with new branch if there is still interest in this module
